### PR TITLE
Redirect HTTP to HTTPS

### PR DESCRIPTION
**Pull request by bersace**

### CODE REVIEW

File: temboardui/autossl.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        "Location: %(new_url)s\r\n"
+        "\r\n" % dict(new_url=new_url)
+    )
+
+
+class AutoHTTPSServer(HTTPServer):
+    # Naïve HTTP server redirecting HTTP requests to HTTPS.
+
+    def handle_stream(self, stream, address):
+        stream.socket.setblocking(1)
```
Comment (bersace) [inline temboardui/autossl.py]: @julmon I'm not happy with that. I will subclass SSHIOStream to implement async handshake.

File: temboardui/autossl.py [hunk 2]
```diff
@@ -0,0 +1,243 @@
+# -*- coding: utf-8 -*-
+#
```
Comment (bersace) [inline temboardui/autossl.py]: @julmon @pgiraud please read the comment, this should help review :)

File: temboardui/autossl.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        lines[:] = lines[:lines.index('')]
+    return (
+        parse_request_start_line(start_line.strip()),
+        HTTPHeaders.parse('\r\n'.join(lines)),
+    )
+
+
+def protocol_switcher(request):
+    new_url = 'https://%s%s' % (
+        request.headers.get('Host', '0.0.0.0:8888'),
```
Comment (julmon) [inline temboardui/autossl.py]: Listening port value shouldn't be hardcoded.

**Comment by bersace** [comment]
@julmon i can't reproduce :x

``` console
$ curl -vv -X GET http://127.0.0.1:8888
Note: Unnecessary use of -X or --request, GET is already inferred.
* Rebuilt URL to: http://127.0.0.1:8888/
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 8888 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8888
> User-Agent: curl/7.52.1
> Accept: */*
>
< HTTP/1.1 301 Moved Permanently
< Content-Length: 0
< Location: https://127.0.0.1:8888/
<
* Curl_http_done: called premature == 0
* Connection #0 to host 127.0.0.1 left intact
```

File: temboardui/autossl.py (hunk 3 above)
Comment (bersace) [inline temboardui/autossl.py]: Fixed.

**Pull request merged**